src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -765,6 +765,21 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetPrimary80215
     return CHIP_NO_ERROR;
 };
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetSlaacIPv6Address(chip::Inet::IPAddress & addr)
+{
+    for (const otNetifAddress * otAddr = otIp6GetUnicastAddresses(mOTInst); otAddr != nullptr; otAddr = otAddr->mNext)
+    {
+        if (otAddr->mValid && otAddr->mAddressOrigin == OT_ADDRESS_ORIGIN_SLAAC)
+        {
+            addr = ToIPAddress(otAddr->mAddress);
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    return CHIP_ERROR_INCORRECT_STATE;
```

Comment:
would NOT_FOUND be a more accurate return code? or is this really about state?

Reply:
yeah, I also felt CHIP_ERROR_INCORRECT_STATE might not be accurate as it suggests an application error, but I couldn't find a generic NOT_FOUND in the error list. Do you think I should add CHIP_ERROR_NOT_FOUND?

Reply:
Yes, I thought one existed (almost any error code list has such a thing, along with things like 'invalid argument' or 'out of range'. I am very surprised we do not have one :( maybe we should add it, since really this is a search that returns not found.

Reply:
I found CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND which seems suitable so I used it here. Let me know if that's better.